### ISSUE TITLE
Integrate smartctl_exporter into the Agent

### ISSUE THREAD

**Issue by DragoonAethis** [grafana/alloy#230]
### Request

Integrate https://github.com/prometheus-community/smartctl_exporter into the unified Grafana Agent binary.

### Use case

SMART is a technology used to collect operational statistics and health indicators for hard drives - it is often the best way to get an early warning that a drive is about to fail. Under Linux, `smartmontools` and `smartctl` can be used to read SMART data and `smartctl_exporter` makes it easy to scrape this data. We need it to monitor the health and status of our physical hard drives.

It would be nice if the exporter was integrated into the Grafana Agent directly, instead of having to install it on each machine. The exporter itself is very limited, as most of its functionality depends on an external binary + database files required for it to work.

**Comment by kyrofa** [comment]
I'm interested in taking this on, but as far as I can see it's a bit unique in that the existing collector relies on smartmontools being installed. Do any other flow components have such a requirement today? Can you provide me a reference? Or should I write a new thing using something like https://github.com/anatol/smart.go?

**Comment by Skaronator** [comment]
In my opinion, it would be best to use the anatol/smart.go library to avoid requiring users to install smartmontools, which is often difficult or impossible on immutable host systems.

**Comment by kyrofa** [comment]
Agreed, I'll have to see how capable it (or other similar things) is.
